lib/hanami/cli.rb

```diff
@@ -131,6 +131,18 @@ def routes
       end
     end
 
+    desc 'secret', 'Prints a secure hash'
+    long_desc <<-EOS
+    `hanami secret` will generate a secure key to be used by the developer as they want, e.g. cookie sessions.
+    EOS
+    def secret(length = '64')
+      require 'securerandom'
+      # secure random generates a string as length * 2.
+      # We want to return the required length by the developer
+      secret = SecureRandom.hex(length.to_i)
```

Comment:
The problem with i / 2 would be that: 

```
number_of_bytes = 31 / 2 # (it results in 15)
secret = SecureRandom.hex(number_of_bytes)
# secret.length would be 30 and not 31
```
After the * operation the odd would return 30 instead of 31.

Did I misunderstand you?